code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java

```diff
@@ -152,7 +156,7 @@ private LdapContext login(String username, String password) {
             userContext = new InitialLdapContext(props, null);
             return userContext;
         } catch (NamingException e) {
-            logger.info("Failed to login to ldap user:" + username, e);
+            logger.info("Failed to login to ldap user:" + username + " " + e.getExplanation());
```

Comment:
It is dependent on context. In this specific case we throw so that it can be caught. Whereas in others it is not. I could refactor it acts differently. but there are many reason why a NamingException is thrown in this specific spot it is generally as a result of an invalid login. In other cases it happens due to the user searching for an invalid user.